morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -78,49 +70,29 @@
 
     private final Morphia morphia;
     private final MongoClient mongoClient;
-    private final DB db;
+    private final MongoDatabase database;
+    private final IndexHelper indexHelper;
+    private DB db;
     private Mapper mapper;
     private WriteConcern defConcern = WriteConcern.ACKNOWLEDGED;
     private DBDecoderFactory decoderFactory;
 
     private volatile QueryFactory queryFactory = new DefaultQueryFactory();
 
-    /**
-     * Create a new DatastoreImpl
-     *
-     * @param morphia     the Morphia instance
-     * @param mongoClient the connection to the MongoDB instance
-     * @param dbName      the name of the database for this data store.
-     */
-    public DatastoreImpl(final Morphia morphia, final MongoClient mongoClient, final String dbName) {
+    DatastoreImpl(final Morphia morphia, final MongoClient mongoClient, final String dbName) {
```

Comment:
So you can deprecate the DatastoreImpl class, but you still shouldn't change any existing constructors.  So can you put this back as is?